components/PageMiddleware/PageMiddleware.tsx

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from "react";
+import useApp from "../../hooks/useApp";
+import { AppProps } from "../../types";
+import jwt from "jsonwebtoken";
+import { isInIframe } from "../../utils/misc";
+import { DashboardTokenPayload } from "../../lib/middlewares";
+import AccessWarning from "../AccessWarning/AccessWarning";
+
+const PageMiddleware = ({ Component, pageProps }: AppProps) => {
```

Comment:
@Cloud11PL Looks good! My only concern is that the word `middleware` is (mostly?) associated with the server-side context. How about we change it to avoid the confusion with the middleware used in the API routes?